subsys/bluetooth/services/fast_pair/fp_crypto/fp_crypto_psa.c

```diff
@@ -0,0 +1,253 @@
+		err = -EIO;
+	}
+
+	status = psa_destroy_key(key_id);
+	if (status != PSA_SUCCESS) {
+		LOG_ERR("psa_destroy_key failed (err: %d)", status);
+	}
+
+	return err;
+}
+
+int fp_crypto_aes128_ecb_encrypt(uint8_t *out, const uint8_t *in, const uint8_t *k)
+{
+	return aes128_ecb_crypt(out, in, k, true);
+}
+
+int fp_crypto_aes128_ecb_decrypt(uint8_t *out, const uint8_t *in, const uint8_t *k)
+{
+	return aes128_ecb_crypt(out, in, k, false);
+}
+
+static psa_key_id_t import_ecdh_priv_key(const uint8_t *data)
+{
+	static const size_t len = 32;
+
+	psa_status_t status;
+	psa_key_id_t key_id = PSA_KEY_ID_NULL;
+	psa_key_attributes_t key_attr = PSA_KEY_ATTRIBUTES_INIT;
+
+	psa_set_key_usage_flags(&key_attr, PSA_KEY_USAGE_DERIVE);
+	psa_set_key_lifetime(&key_attr, PSA_KEY_LIFETIME_VOLATILE);
+	psa_set_key_algorithm(&key_attr, PSA_ALG_ECDH);
+	psa_set_key_type(&key_attr, PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_SECP_R1));
+	psa_set_key_bits(&key_attr, len * CHAR_BIT);
+
+	status = psa_import_key(&key_attr, data, len, &key_id);
+	psa_reset_key_attributes(&key_attr);
+
+	if (status != PSA_SUCCESS) {
+		LOG_ERR("psa_import_key failed (err: %d)", status);
+		key_id = PSA_KEY_ID_NULL;
+	}
+
+	return key_id;
+}
+
+int fp_crypto_ecdh_shared_secret(uint8_t *secret_key, const uint8_t *public_key,
+				 const uint8_t *private_key)
+{
+	/* Marker of the uncompressed binary format for a point on an elliptic curve. */
+	static const uint8_t uncompressed_format_marker = 0x04;
+
+	uint8_t public_key_uncompressed[FP_CRYPTO_ECDH_PUBLIC_KEY_LEN +
+					sizeof(uncompressed_format_marker)];
+	int err = 0;
+	size_t out_len = 0;
+	psa_key_handle_t priv_key_id;
+	psa_status_t status;
+
+	priv_key_id = import_ecdh_priv_key(private_key);
+	if (priv_key_id == PSA_KEY_ID_NULL) {
+		LOG_ERR("import_ecdh_shared_secret_key failed");
+		return -EIO;
+	}
+
+	/* Use the uncompressed binary format [0x04 X Y] for the public key point. */
+	public_key_uncompressed[0] = uncompressed_format_marker;
+	memcpy(&public_key_uncompressed[1], public_key, FP_CRYPTO_ECDH_PUBLIC_KEY_LEN);
+
+	status = psa_raw_key_agreement(PSA_ALG_ECDH, priv_key_id,
+				       public_key_uncompressed, sizeof(public_key_uncompressed),
+				       secret_key, FP_CRYPTO_ECDH_SHARED_KEY_LEN, &out_len);
+	if (status != PSA_SUCCESS) {
+		LOG_ERR("psa_raw_key_agreement failed (err: %d)", status);
+		err = -EIO;
+	}
+
+	if (!err && (out_len != FP_CRYPTO_ECDH_SHARED_KEY_LEN)) {
+		LOG_ERR("Invalid psa_raw_key_agreement output len: %zu", out_len);
+		err = -EIO;
+	}
+
+	status = psa_destroy_key(priv_key_id);
+	if (status != PSA_SUCCESS) {
+		LOG_ERR("psa_destroy_key failed (err: %d)", status);
+	}
+
+	return err;
+}
+
+static int fp_crypto_psa_init(void)
+{
+	psa_status_t status = psa_crypto_init();
+
+	if (status != PSA_SUCCESS) {
+		LOG_ERR("psa_crypto_init failed (err: %d)", status);
+		k_panic();
+		return -EIO;
+	}
+
+	return 0;
+}
+
+SYS_INIT(fp_crypto_psa_init, APPLICATION, CONFIG_APPLICATION_INIT_PRIORITY);
```

Comment:
As discussed before, I opened a follow-up ticket (`NCSDK-26882`). Since we removed the ticket from scope of current sprint, it will be added after this PR is merged.